presto-native-execution/presto_cpp/main/common/Configs.cpp

```diff
@@ -25,12 +28,135 @@ namespace fs = std::experimental::filesystem;
+
+void checkIncomingSystemProperties(
+    const std::unordered_map<std::string, std::string>& values) {
+  static const std::unordered_set<std::string_view> kSupportedSystemProperties{
+      SystemConfig::kMutableConfig,
+      SystemConfig::kPrestoVersion,
+      SystemConfig::kHttpServerHttpPort,
+      SystemConfig::kHttpServerReusePort,
+      SystemConfig::kDiscoveryUri,
+      SystemConfig::kMaxDriversPerTask,
+      SystemConfig::kConcurrentLifespansPerTask,
+      SystemConfig::kHttpExecThreads,
+      SystemConfig::kHttpServerHttpsPort,
+      SystemConfig::kHttpServerHttpsEnabled,
+      SystemConfig::kHttpsSupportedCiphers,
+      SystemConfig::kHttpsCertPath,
+      SystemConfig::kHttpsKeyPath,
+      SystemConfig::kHttpsClientCertAndKeyPath,
+      SystemConfig::kNumIoThreads,
+      SystemConfig::kNumQueryThreads,
+      SystemConfig::kNumSpillThreads,
+      SystemConfig::kSpillerSpillPath,
+      SystemConfig::kShutdownOnsetSec,
+      SystemConfig::kSystemMemoryGb,
+      SystemConfig::kAsyncCacheSsdGb,
+      SystemConfig::kAsyncCacheSsdCheckpointGb,
+      SystemConfig::kAsyncCacheSsdPath,
+      SystemConfig::kAsyncCacheSsdDisableFileCow,
+      SystemConfig::kEnableSerializedPageChecksum,
+      SystemConfig::kUseMmapArena,
+      SystemConfig::kMmapArenaCapacityRatio,
+      SystemConfig::kUseMmapAllocator,
+      SystemConfig::kEnableVeloxTaskLogging,
+      SystemConfig::kEnableVeloxExprSetLogging,
+      SystemConfig::kLocalShuffleMaxPartitionBytes,
+      SystemConfig::kShuffleName,
+      SystemConfig::kHttpEnableAccessLog,
+      SystemConfig::kHttpEnableStatsFilter,
+      SystemConfig::kRegisterTestFunctions,
+      SystemConfig::kHttpMaxAllocateBytes,
+      SystemConfig::kQueryMaxMemoryPerNode,
+  };
+
+  std::stringstream supported;
+  std::stringstream unsupported;
+  for (const auto& pair : values) {
+    ((kSupportedSystemProperties.count(pair.first) != 0) ? supported
+                                                         : unsupported)
+        << "  " << pair.first << "=" << pair.second << "\n";
+  }
+  LOG(INFO) << "STARTUP: Supported system properties:\n" << supported.str();
+  LOG(INFO) << "STARTUP: Unsupported system properties:\n" << unsupported.str();
+}
+
+enum class CapacityUnit {
+  BYTE,
+  KILOBYTE,
+  MEGABYTE,
+  GIGABYTE,
+  TERABYTE,
+  PETABYTE
+};
+
+double toBytesPerCapacityUnit(CapacityUnit unit) {
+  switch (unit) {
+    case CapacityUnit::BYTE:
+      return 1;
+    case CapacityUnit::KILOBYTE:
+      return exp2(10);
+    case CapacityUnit::MEGABYTE:
+      return exp2(20);
+    case CapacityUnit::GIGABYTE:
+      return exp2(30);
+    case CapacityUnit::TERABYTE:
+      return exp2(40);
+    case CapacityUnit::PETABYTE:
+      return exp2(50);
+    default:
+      VELOX_USER_FAIL("Invalid capacity unit '{}'", (int)unit);
+  }
+}
+
+CapacityUnit valueOfCapacityUnit(const std::string& unitStr) {
+  if (unitStr == "B") {
+    return CapacityUnit::BYTE;
+  }
+  if (unitStr == "kB") {
```

Comment:
Why is this the only one lowercase 'k'?

Reply:
Copied from the protocol code, which based on Java Presto code, which uses lower case.
It is probably the standard for kb.